src/ds/stacktrace_utils.h

```diff
@@ -27,12 +27,78 @@ namespace stacktrace
 #endif
   }
 
+#ifndef INSIDE_ENCLAVE
+  static void sig_handler(int signo, siginfo_t* info, void* _ctx)
+  {
+    backward::SignalHandling::handleSignal(signo, info, _ctx);
+    LOG_FATAL_FMT("Handled fatal signal {}", signo);
+
+    // SA_RESETHAND seems to be insufficient, so we manually reset the handler
+    // before re-raising
+    signal(signo, SIG_DFL);
+    raise(signo);
+  }
+#endif
+
   static inline void init_sig_handlers()
   {
 #ifndef INSIDE_ENCLAVE
-    // This registers signal handlers on construction, and these remain after
-    // destruction.
-    backward::SignalHandling sh;
+    // This is based on the constructor of backward::SignalHandling, but avoids
+    // infinitely recursing stacktraces
+    constexpr size_t stack_size = 1024 * 1024 * 8;
```

Comment:
We typically run with 1024 stack pages, so unless I'm missing something, this will turn out to be too much in most of our SGX tests?